src/Simplex/Messaging/Agent/Store.hs

```diff
@@ -46,14 +46,20 @@ class Monad m => MonadAgentStore s m where
   setRcvQueueStatus :: s -> RcvQueue -> QueueStatus -> m ()
   setRcvQueueActive :: s -> RcvQueue -> VerificationKey -> m ()
   setSndQueueStatus :: s -> SndQueue -> QueueStatus -> m ()
+  updateSignatureKey :: s -> SndQueue -> SignatureKey -> m ()
+
+  -- Confirmations
+  createConfirmation :: s -> TVar ChaChaDRG -> NewConfirmation -> m ConfirmationId
+  getConfirmation :: s -> ConfirmationId -> m Confirmation
+  approveConfirmation :: s -> ConfirmationId -> ConnInfo -> m ()
```

Comment:
I don't see any situation where getConfirmation and approveConfirmation are used separately, so it could be just approveConfirmation that returns the confirmation.

Reply:
done